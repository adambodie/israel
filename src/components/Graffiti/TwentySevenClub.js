import React, { Component } from 'react'
import { Column } from 'react-foundation'
import Modal from 'react-modal'

const customStyles = {
	content : {
		top: '10%',
		left: '50%',
		right: 'auto',
		bottom: 'auto',
		marginRight: '-50%',
		transform: 'translate(-50%, -50%)'
	}
}

export default class TwentySevenClub extends Component {  
	constructor() {
		super()
		this.state = {
			members: [
				{name: 'Brian Jones', className: 'one'},
				{name: 'Jimi Hendrix', className: 'two'},
				{name: 'Janis Joplin', className: 'three'},
				{name: 'Jim Morrison', className: 'four'},
				{name: 'Basquiat', className: 'five'},
				{name: 'Kurt Cobain', className: 'six'},
				{name: 'Amy Winehouse', className: 'seven'},
				{name: 'Kis Lev?', className: 'eight'},
			]
		}
		this.openModal = this.openModal.bind(this)
		this.closeModal = this.closeModal.bind(this)
	}  
	openModal(e, index) {
		const name = this.state.members[index].name
		this.setState({
			modalIsOpen: true,
			name: name,
		})
	}
	closeModal() {
		this.setState({modalIsOpen: false})
	}
	render() {
		const { members, name } = this.state
		return (
			<Column>
				<h1>The 27 Club</h1>
				<div className='club'>
					<img src='/assets/main/27-club.jpg' alt='27 Club' className='club-image'/>
					{members.map((x, index) => <button key={index} data-testid={`club-cover cover-${x.className}`} className={`club-cover cover-${x.className}`} onClick={() => this.openModal(this, index)}></button>)}
				</div>
				<p>The 27 Club is an informal club of famous artists who all passed away at the age of 27 years old.  Click on each person to see which famous artist is depicted.  The last person on the right is the creator of the mural, graffiti artist Jonathan Kis-Lev, who came up with the idea of this mural at 26 years old, fearing he'd join the club.  There are rumors regarding why his face is covered in pink paint and who covered it in pink. Kis-Lev, with the help of other artists, painted this within two days, with the use of the crane they rented.  In order to avoid being caught, they completed the mural on Shabbat, claiming a permit, to keep police from calling to check on the permit they didn't have.</p>
				<Modal
					isOpen={this.state.modalIsOpen}
					onRequestClose={this.closeModal}
					style={customStyles}
					contentLabel='Twenty Seven Club Modal'
					appElement={document.getElementById('root')}
				>
					<h2>{name}</h2>
					<button className='shakshuka-button' onClick={this.closeModal}>x</button>
				</Modal>
			</Column>
		)
	}
}
