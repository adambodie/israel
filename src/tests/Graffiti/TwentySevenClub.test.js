import React from 'react'
import renderer from 'react-test-renderer'
import {
  render,
  cleanup,
  fireEvent
} from 'react-testing-library'

import TwentySevenClub from '../../components/Graffiti/TwentySevenClub'

afterEach(cleanup) 

it('renders correctly', () => {
	const tree = renderer.create(<TwentySevenClub />).toJSON()
	expect(tree).toMatchSnapshot()
})

it('opens a modal and closes it', () => {
	const spyOpen = jest.spyOn(TwentySevenClub.prototype, 'openModal')
	const spyClose = jest.spyOn(TwentySevenClub.prototype, 'closeModal')
	const { getByText, getByTestId, rerender } = render(<TwentySevenClub />)
	fireEvent.click(getByTestId('club-cover cover-one'))
	expect(spyOpen).toHaveBeenCalled()
	rerender(<TwentySevenClub />)
	fireEvent.click(getByText('x'))
	expect(spyClose).toHaveBeenCalled()
})