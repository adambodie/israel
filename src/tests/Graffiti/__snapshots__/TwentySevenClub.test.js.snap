// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="columns"
>
  <h1>
    The 27 Club
  </h1>
  <div
    className="club"
  >
    <img
      alt="27 Club"
      className="club-image"
      src="/assets/main/27-club.jpg"
    />
    <button
      className="club-cover cover-one"
      data-testid="club-cover cover-one"
      onClick={[Function]}
    />
    <button
      className="club-cover cover-two"
      data-testid="club-cover cover-two"
      onClick={[Function]}
    />
    <button
      className="club-cover cover-three"
      data-testid="club-cover cover-three"
      onClick={[Function]}
    />
    <button
      className="club-cover cover-four"
      data-testid="club-cover cover-four"
      onClick={[Function]}
    />
    <button
      className="club-cover cover-five"
      data-testid="club-cover cover-five"
      onClick={[Function]}
    />
    <button
      className="club-cover cover-six"
      data-testid="club-cover cover-six"
      onClick={[Function]}
    />
    <button
      className="club-cover cover-seven"
      data-testid="club-cover cover-seven"
      onClick={[Function]}
    />
    <button
      className="club-cover cover-eight"
      data-testid="club-cover cover-eight"
      onClick={[Function]}
    />
  </div>
  <p>
    The 27 Club is an informal club of famous artists who all passed away at the age of 27 years old.  Click on each person to see which famous artist is depicted.  The last person on the right is the creator of the mural, graffiti artist Jonathan Kis-Lev, who came up with the idea of this mural at 26 years old, fearing he'd join the club.  There are rumors regarding why his face is covered in pink paint and who covered it in pink. Kis-Lev, with the help of other artists, painted this within two days, with the use of the crane they rented.  In order to avoid being caught, they completed the mural on Shabbat, claiming a permit, to keep police from calling to check on the permit they didn't have.
  </p>
</div>
`;
